bin/resty

```diff
@@ -368,6 +368,54 @@ sub sigint {
 }
 $SIG{INT} = \&sigint;
 
+sub sigterm {
+    $SIG{TERM} = \&sigterm;
+    if ($child_pid) {
+        kill TERM => $child_pid;
+    }
+}
+$SIG{TERM} = \&sigterm;
+
+sub sigquit {
+    $SIG{QUIT} = \&sigquit;
+    if ($child_pid) {
+        kill QUIT => $child_pid;
+    }
+}
+$SIG{QUIT} = \&sigquit;
+
+sub sighup {
+    $SIG{HUP} = \&sighup;
+    if ($child_pid) {
+        kill HUP => $child_pid;
+    }
+}
+$SIG{HUP} = \&sighup;
+
+sub sigusr1 {
+    $SIG{USR1} = \&sigusr1;
+    if ($child_pid) {
+        kill USR1 => $child_pid;
+    }
+}
+$SIG{USR1} = \&sigusr1;
+
+sub sigusr2 {
+    $SIG{USR2} = \&sigusr2;
+    if ($child_pid) {
+        kill USR2 => $child_pid;
+    }
+}
+$SIG{USR2} = \&sigusr2;
+
+sub sigwinch {
+    $SIG{WINCH} = \&sigwinch;
+    if ($child_pid) {
+        kill WINCH => $child_pid;
+    }
+}
+$SIG{WINCH} = \&sigwinch;
```

Comment:
It seems like you can just use a loop to avoid code duplication here? Something like (untested)

```perl
for my $sig (qw/ TERM QUIT HUP USR1 USR2 WINCH /) {
    $SIG{$sig} = sub {
        if ($child_pid) {
            kill $sig => $child_pid;
        }
    };
}
```